frontend/src/scenes/dashboard/dashboardLogic.js

```diff
@@ -138,12 +150,68 @@ export const dashboardLogic = kea({
         saveLayouts: async (_, breakpoint) => {
             await breakpoint(300)
             await api.update(`api/dashboard_item/layouts`, {
-                items: values.items.map(item => ({ id: item.id, layouts: item.layouts })),
+                items: values.items.map(item => {
+                    const layouts = {}
+                    Object.entries(item.layouts).forEach(([key, layout]) => {
+                        const { i, ...rest } = layout
+                        layouts[key] = rest
+                    })
+                    return { id: item.id, layouts }
+                }),
             })
         },
 
         updateItemColor: ({ id, color }) => {
             api.update(`api/dashboard_item/${id}`, { color })
         },
+
+        duplicateDashboardItem: async ({ id, dashboardId, move }) => {
+            const item = values.items.find(item => item.id === id)
+            if (item) {
+                const { id: _discard, ...rest } = item
+                const newItem = dashboardId ? { ...rest, dashboard: dashboardId } : { ...rest }
+                const addedItem = await api.create('api/dashboard_item', newItem)
+
+                if (move) {
+                    const deletedItem = await api.update(`api/dashboard_item/${item.id}`, { deleted: true })
+                    dashboardsModel.actions.updateDashboardItem(deletedItem)
+
+                    const toastId = toast(
+                        <div>
+                            Panel moved to dashboard.&nbsp;
```

Comment:
Would be nice to have a link to that dashboard here

Reply:
added!